crest/Assets/Crest/Crest/Scripts/Shapes/FFT/ShapeFFT.cs

```diff
@@ -95,6 +97,18 @@ public partial class ShapeFFT : MonoBehaviour, LodDataMgrAnimWaves.IShapeUpdatab
         [Tooltip("Maximum amount a point on the surface will be displaced horizontally by waves from its rest position. Increase this if gaps appear at sides of screen."), SerializeField]
         float _maxHorizontalDisplacement = 15f;
 
+        [Header("Collision Data Baking")]
+        [Tooltip("Enable running this FFT with baked data. This makes the FFT periodic (repeating in time).")]
+        public bool _enableBakedCollision = false;
+        [Tooltip("Frames per second of baked data. Larger values generate more frames and increase baked data size."), DecoratedField, Predicated("_enableBakedCollision")]
+        public int _timeResolution = 4;
+        [Tooltip("Smallest wavelength required in collision. To preview disable power sliders in spectrum for smaller values than this number. Smaller values require more resolution and increase baked data size."), DecoratedField, Predicated("_enableBakedCollision")]
+        public float _smallestWavelengthRequired = 2f;
+        [Tooltip("FFT waves will loop with a periodic of this many seconds."), DecoratedField, Predicated("_enableBakedCollision"), Range(4f, 128f), Delayed]
```

Comment:
Cannot mix DecoratedField, Range and Delayed together as they are fields attributes rather than decorators (like Predicated). You'll need to choose either Ranged or Delayed.

I should probably make Delayed a decorator.

Reply:
Yeah making a Delayed decorator might be useful as this wants both delayed and range.

I removed delayed and noticed ranged is not working (no slider in gui), can you spot what it needs? I think i've bumped into this before but can never remember what to do to get it working. If this issue could be caught with an error or assert or such that would be helpful.

Reply:
DecoratedField needs to be removed. Both Range and DecoratedField are fields and there can be only one. DecoratedField exists only to make decorators work. Range is already a decorated field so decorators work if that makes sense.

> If this issue could be caught with an error or assert or such that would be helpful.

Good idea. I'll look into adding it.

Reply:
ah thanks. fixed now. sounds good